src/core_plugins/elasticsearch/lib/set_headers.js

```diff
@@ -0,0 +1,15 @@
+import { isObject } from 'lodash';
+
+export default function setHeaders(originalHeaders, newHeaders) {
+  if (!isObject(originalHeaders)) {
```

Comment:
I agree with @ycombinator here. If the intention is to verifying that the arguments are compatible with the spread operator then so be it, but if the intention is to make sure that `setHeaders()` is being used as intended, a more strict check is probably appropriate.